### PR TITLE
Switch nav. drawer to new NavigationView class (don't merge yet)

### PR DESCRIPTION
**Pull request by timrae**
I had a go at switching the navigation drawer over to the new [`NavigationView`](https://developer.android.com/reference/android/support/design/widget/NavigationView.html) convenience class in the Android design support library. It works fine, cleans up the `NavigationDrawerActivity` code nicely, and it will be [super easy to add our branding to the top of the drawer](http://stackoverflow.com/questions/26440879/how-do-i-use-drawerlayout-to-display-over-the-actionbar-toolbar-and-under-the-st/30525387#30525387) this way. 

Unfortunately, Google have not given developers the ability to add custom items to the drawer... A developer from Google said "NavigationView is designed to meet the material design guidelines so intentionally breaking the guidelines isn't a use case﻿"... Which, to be honest is kind of a retarded thing to say because  the Gmail app uses custom views in the nav. drawer to show the number of counts for each label, and the Play Movies app has a switch to disable showing movies which aren't downloaded yet. So there clearly are usage cases. When I mentioned this I got no response; I don't know if that's a good thing or a bad thing. I also [asked on stackoverflow](http://stackoverflow.com/questions/30728842/how-can-i-add-a-custom-item-to-a-navigationview-with-a-menu-layout) and nobody replied

In our specific case I'd like to move the night mode switch there in the event that someone gets around to adding a global night-mode theme (e.g. @jshevek showed some interest in this in the past). Of course we *could* add a night-mode menu item to every activity in the app instead of putting a switch in the nav. drawer, but that's kind of ugly, so I'm not sure if merging this is a good idea yet. Thoughts?

Maybe in a future version of the design library Google will allow custom views, or once they release the source code we can modify the class to allow this, or a 3rd party library will make it easy for us.

TL:DR 
Upside of merging this: 1) cleaner code, 2) ability to easily add branding (I can't really be bothered adding it to our current layout)
Downside of merging this: 1) not futureproof as we can't add custom Views like a switch

### CODE REVIEW

**Comment by jshevek** [comment]
Regarding the loss of the switch:  Maybe the icon on the left can be used to indicate the night mode state, instead of a switch?  

(App then loses the switch as a visual hint indicating the action that results from clicking that item, so the text may need to be more explicit to compensate for this, like "Toggle Night Mode" vs "Night Mode")

Also, someone has responded on SO since your last post.  I'm not sure if their suggestion solves the future proofing problem.

**Comment by timrae** [comment]
Nice idea, I like it. I just gave it a try, and we can easily dynamically change the icon when the drawer is created, and when the night mode item is clicked:

``` java
    private void setNightModeIcon() {
        Menu menu = mNavigationView.getMenu();
        MenuItem item = menu.findItem(R.id.nav_night_mode);
        if (item != null) {
            SharedPreferences prefs = AnkiDroidApp.getSharedPrefs(this);
            boolean nightMode = prefs.getBoolean("invertedColors", false);
            if (nightMode) {
                item.setIcon(R.drawable.ic_night_mode_on);
            } else {
                item.setIcon(R.drawable.ic_night_mode_off);
            }
        } else {
            Timber.e("Could not find night-mode MenuItem");
        }
    }
```

**Comment by timrae** [comment]
We could easily make a new switch icon, or we could use moon/sun icons like we have now... In any case, we can think about that if/when we have a system-wide dark theme. It's feasible that the support library will be updated to allow custom views in the future as well.

How about merging this?
